tests/Unit/ApparentHorizons/Test_StrahlkorperDataBox.cpp

```diff
@@ -0,0 +1,285 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+
+#include "ApparentHorizons/SpherepackIterator.hpp"
+#include "ApparentHorizons/Strahlkorper.hpp"
+#include "ApparentHorizons/StrahlkorperDataBox.hpp"
+#include "DataStructures/DataBox.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "tests/Unit/ApparentHorizons/YlmTestFunctions.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace {
+
+void test_radius_and_derivs() {
+  {
+    // Create spherical Strahlkorper
+    const std::array<double, 3> center = {{0.1, 0.2, 0.3}};
+    const double r = 3.0;
+    Strahlkorper<Frame::Inertial> s(4, 4, r, center);
+    CHECK(s.average_radius() == approx(r));
+  }
+
+  // Create a strahlkorper with a Im(Y11) dependence.
+  const size_t l_max = 4, m_max = 4;
+  const double radius = 2.0;
+  const double y11_amplitude = 1.0;
+  const std::array<double, 3> center = {{0.1, 0.2, 0.3}};
+
+  Strahlkorper<Frame::Inertial> strahlkorper_sphere(l_max, m_max, radius,
+                                                    center);
+  auto coefs = strahlkorper_sphere.coefficients();
+  SpherepackIterator it(l_max, m_max);
+  // Conversion between SPHEREPACK b_lm and real valued harmonic coefficients:
+  // b_lm = (-1)^{m+1} sqrt(1/2pi) d_lm
+  coefs[it.set(1, -1)()] = y11_amplitude * sqrt(0.5 / M_PI);
+  Strahlkorper<Frame::Inertial> strahlkorper(coefs, strahlkorper_sphere);
+
+  // Now construct a Y00 + Im(Y11) surface by hand.
+  const auto& theta_phi = strahlkorper.ylm_spherepack().theta_phi_points();
+  const auto& theta_points = strahlkorper.ylm_spherepack().theta_points();
+  const auto& phi_points = strahlkorper.ylm_spherepack().phi_points();
+  const auto n_pts = theta_phi[0].size();
+  YlmTestFunctions::Y11 func;
+  DataVector test_radius(n_pts);
+  func.func(&test_radius, 1, 0, theta_points, phi_points);
```

Comment:
our convention is to use `expected_radius` instead of `test_radius`  (please apply this to all such uses below)